source/Octopus.Cli/Commands/Releases/UnblockReleaseCommand.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Threading.Tasks;
+using Octopus.Cli.Infrastructure;
+using Octopus.Cli.Repositories;
+using Octopus.Cli.Util;
+using Octopus.Client;
+using Octopus.Client.Exceptions;
+using Octopus.Client.Model;
+
+namespace Octopus.Cli.Commands.Releases
+{
+    [Command("unblock-release", "resume-release-progression", Description = "Unblock a release from progressing to next phase.")]
+    public class UnblockReleaseCommand : ApiCommand, ISupportFormattedOutput
+    {
+        ProjectResource project;
+        ReleaseResource release;
+
+        public UnblockReleaseCommand(IOctopusClientFactory clientFactory, IOctopusAsyncRepositoryFactory repositoryFactory, IOctopusFileSystem fileSystem, ICommandOutputProvider commandOutputProvider)
+            : base(clientFactory, repositoryFactory, fileSystem, commandOutputProvider)
+        {
+            var options = Options.For("Unblocking release.");
+            options.Add("project=", "Name or ID of the project", v => ProjectNameOrId = v);
+            options.Add("version=|releaseNumber=", "Release version/number.",
+                v => ReleaseVersionNumber = v);
+        }
+
+        public string ProjectNameOrId { get; set; }
+
+        public string ReleaseVersionNumber { get; set; }
+
+        protected override async Task ValidateParameters()
+        {
+            if (string.IsNullOrWhiteSpace(ProjectNameOrId)) throw new CommandException("Please specify a project name or ID using the parameter: --project=XYZ");
+            if (string.IsNullOrWhiteSpace(ReleaseVersionNumber)) throw new CommandException("Please specify a release version number using the version parameter: --version=1.0.5");
+            if (!SemanticVersion.TryParse(ReleaseVersionNumber, out _)) throw new CommandException("Please provide a valid release version format, you can refer to https://semver.org/ for a valid format: --version=1.0.5");
+
+            await base.ValidateParameters().ConfigureAwait(false);
+        }
+
+        public async Task Request()
+        {
+            project = await Repository.Projects.FindByNameOrIdOrFail(ProjectNameOrId).ConfigureAwait(false);
+
+            release = await Repository.Projects.GetReleaseByVersion(project, ReleaseVersionNumber).ConfigureAwait(false);
+            if (release == null) throw new OctopusResourceNotFoundException($"Unable to locate a release with version/release number '{ReleaseVersionNumber}'.");
+
+            await Repository.Defects.ResolveDefect(release).ConfigureAwait(false);
```

Comment:
Should we check whether the defect is already resolved to avoid problems where someone already resolved it manually? E.g. https://build.octopushq.com/viewLog.html?buildId=1262124&buildTypeId=Octofront_HostedE2eTests.  Same question for the Block/Prevent command.

Reply:
I guess this is where part of the confusion coming from. From preventing/allowing release progression perspective, it makes sense to not throw exception but just inform users that the release prevention/allowance is already set. However, from the fundamental api endpoint perspective, it makes sense to throw exception as that is what we initially design that endpoint for. 

I personally prefer to still throw exception, but I would catch the exception and turn it into something else like "Release with verions '0.0.1' is already prevented", otherwise the following exception message is confusing to people ... in a way when they are using a command `prevent-release-progression`

![image](https://user-images.githubusercontent.com/12688884/76283161-9a101700-62e5-11ea-8da2-0c3cbd785636.png)

Reply:
However, it's using a general exception type `OctopusValidationException`, it doesn't seem to provide much info to help me determine what exactly the exception is apart from the message ... it will be danger to just capture the type ...

I will add the flag for now ...